### PR TITLE
Finalize providers

### PR DESCRIPTION
**Pull request by unkind**
See #537 for details.

This PR is not ready for merge, `GoogleMapsBusiness` tests were not ported yet.

### CODE REVIEW

File: src/Geocoder/Provider/GoogleMaps.php [hunk 1]
```diff
@@ -50,6 +50,38 @@ class GoogleMaps extends AbstractHttpProvider implements LocaleAwareProvider
...
+     * @param HttpClient $client An HTTP adapter
+     * @param string     $clientId Your Client ID
+     * @param string     $privateKey Your Private Key (optional)
+     * @param string     $locale A locale (optional)
+     * @param string     $region Region biasing (optional)
+     * @param bool       $useSsl Whether to use an SSL connection (optional)
+     * @param string     $apiKey Google Geocoding API key (optional)
+     * @return GoogleMaps
+     */
+    public function business(HttpClient $client, $clientId, $privateKey = null, $locale = null, $region = null, $useSsl = false, $apiKey = null)
```
Comment (Nyholm) [inline src/Geocoder/Provider/GoogleMaps.php]: Is this a factory method? Should it be static?

```php
$gBusiness = GoogleMaps::business(...);
```

**Comment by Nyholm** [comment]
Good job. I like this PR. Let me know when you done with your changes.

File: src/Geocoder/Provider/GoogleMaps.php (hunk 1 above)
Comment (unkind) [inline src/Geocoder/Provider/GoogleMaps.php]: Sure.

**Comment by willdurand** [comment]
I like it too.

**Comment by unkind** [comment]
I think this PR is done.

**Comment by willdurand** [comment]
:+1:

**Comment by Nyholm** [comment]
Thank you @unkind. Great work!

**Comment by unkind** [comment]
We shouldn't forget to fix the bundle: https://github.com/geocoder-php/BazingaGeocoderBundle

**Pull request merged**